raster/src/main/scala/geotrellis/raster/ArrayMultibandTile.scala

```diff
@@ -428,4 +428,20 @@ class ArrayMultibandTile(_bands: Array[Tile]) extends MultibandTile with MacroMu
 
     new ArrayMultibandTile(newBands)
   }
+
+  override def equals(other: Any): Boolean = {
+    if (other == null) {
+      false
+    } else other match {
+      case that : ArrayMultibandTile =>
+        var i = 0
+        while (i < bandCount) {
+          if (band(i) != that.band(i)) return false
+          i += 1
+        }
+        true
```

Comment:
Would prefer yhe `cfor` method (cfor is macro which unrolls to structure which is basically as fast as while loop). Loop condition can be modified to break if result becomes false.